### ISSUE TITLE
LazyFrame.collect(background=False) not narrowed to DataFrame with complex overloads

### ISSUE THREAD

**Issue by gaborbernat** [astral-sh/ty#3141]
## Summary

ty fails to narrow the return type of `LazyFrame.collect(background=False)` to `DataFrame` even though a matching `Literal[False]` overload is defined. This occurs when calling a method with an explicit keyword argument matching a Literal overload in complex function signatures with many parameters.

### Minimal Reproducible Example

```python
import polars as pl

lf = pl.LazyFrame({"a": [1, 2, 3]})
result: pl.DataFrame = lf.collect(background=False)  # Error
```

**Error:**
```
invalid-assignment: Object of type InProcessQuery | DataFrame is not assignable to DataFrame
```

### Command Invoked
`ty check` (via tox)

### ty Settings
No custom ty settings. Standard strict type checking enabled.

### Expected Behavior

When calling a method with an explicit keyword argument matching a `Literal` overload, ty should narrow the return type accordingly. The call `lf.collect(background=False)` explicitly matches the overload:

```python
@overload
def collect(
    self,
    *,
    type_coercion: bool = True,
    predicate_pushdown: bool = True,
    # ... many more parameters with defaults
    background: Literal[False] = False,
    # ... more parameters
) -> DataFrame: ...
```

Therefore, `result` should be narrowed to `DataFrame`, not `InProcessQuery | DataFrame`.

### Actual Behavior

ty returns a union type (`InProcessQuery | DataFrame`) instead of narrowing to `DataFrame`, even though:
1. The `background=False` argument is explicitly passed as a keyword argument
2. A matching overload with `Literal[False]` exists
3. The overload clearly returns `DataFrame`

### Pattern

The issue appears when:
1. Multiple overloads have many parameters with defaults
2. The discriminating parameter uses `Literal[False]` with a default value  
3. The implementation has decorators (`@deprecate_streaming_parameter`, `@forward_old_opt_flags`)
4. All other parameters before the discriminating one also have defaults

This works fine with simple overloads but fails with complex polars-like signatures.

### Reproduction Steps

1. Use polars 1.39.0 (or any version with current LazyFrame.collect overloads)
2. Call `LazyFrame.collect(background=False)` with an explicit type annotation expecting `DataFrame`
3. Run `ty check`

### Related Code

- Affected library: [polars LazyFrame.collect](https://github.com/pola-rs/polars/blob/main/py-polars/src/polars/lazyframe/frame.py#L2196)
- Reported in context of: [arh-cli ty migration](https://github.com/bbg-dev/arh-cli/tree/ty)

## Version

Tested and confirmed with:
- ty 0.0.24
- ty 0.0.25 (issue still present)

**Comment by gaborbernat** [comment]
## Minimal Reproducible Example

Created test file showing the issue:

```python
import polars as pl

lf = pl.LazyFrame({"a": [1, 2, 3]})
result: pl.DataFrame = lf.collect(background=False)  # Error: InProcessQuery | DataFrame not assignable to DataFrame
```

The issue occurs with polars 1.39.0. The overloads are properly defined with:
- First overload: `background: Literal[True]` -> `InProcessQuery`
- Second overload: `background: Literal[False] = False` -> `DataFrame`

Even when explicitly passing `background=False`, ty doesn't narrow to the `DataFrame` overload and returns the union type instead.

**Theory**: The implementation has decorators that may interfere with overload resolution, or ty's overload matching has limitations with complex signatures containing many parameters with defaults.

**Comment by gaborbernat** [comment]
This is a duplicate of #2278. The root cause is that polars.LazyFrame.collect() has decorators on the implementation, which breaks overload resolution in ty.

See #2278 for the core issue about decorated functions losing overload information.
